### PR TITLE
Update Service Provider

### PR DESCRIPTION
**Pull request by michaelshaffer37**
This PR updates the service provider so that it sets the DynamoDbClientService on the DynamoDbModel in the boot method similar to the way the `$resolver` and the `$dispatcher` are set on the [Laravel Eloquent Model](https://github.com/laravel/framework/blob/5.4/src/Illuminate/Database/DatabaseServiceProvider.php).

---
This change is [Reviewable](https://reviewable.io/reviews/baopham/laravel-dynamodb/45)

### CODE REVIEW

File: tests/DynamoDbClientServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Class DynamoDbClientServiceTest
+ */
+class DynamoDbClientServiceTest extends TestCase
```
Comment (michaelshaffer37) [inline tests/DynamoDbClientServiceTest.php]: Well I can add a PSR-4 namespace to the whole test dir but I didn't really think that was in the scope of what was trying to be achieved here.

File: tests/DynamoDbClientServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    /**
+     * Make sure we are not leaving any values set on the DynamoDbModel
+     */
+    public function tearDown()
+    {
+        parent::tearDown();
+
+        \BaoPham\DynamoDb\DynamoDbModel::unsetDynamoDbClientService();
+    }
+}
```
Comment (michaelshaffer37) [inline tests/DynamoDbClientServiceTest.php]: Fixed this

File: tests/ModelTest.php [hunk 3]
```diff
@@ -52,6 +53,9 @@ protected function bindDynamoDbClientInstance()
         $this->dynamoDb = new DynamoDbClientService($config, new Marshaler($marshalerOptions),
             new EmptyAttributeFilter);
 
+        // Set the DynamoDbClient, this is handled by the SynamoDbServiceProvider boot in normal use.
```
Comment (baopham) [inline tests/ModelTest.php]: typo `SynamoDbServiceProvider`? :)

File: tests/DynamoDbClientServiceTest.php (hunk 1 above)
Comment (baopham) [inline tests/DynamoDbClientServiceTest.php]: ah yeah, I can do that.

**Comment by baopham** [comment]
Very nice! thank you for the contribution. Once you fix the typo, I'll merge it in 👍

File: tests/ModelTest.php (hunk 3 above)
Comment (michaelshaffer37) [inline tests/ModelTest.php]: Thx for catching that.

File: tests/DynamoDbClientServiceTest.php (hunk 1 above)
Comment (michaelshaffer37) [inline tests/DynamoDbClientServiceTest.php]: I already did the work, just thought it should go on a separate PR.

Comment (baopham) [inline tests/DynamoDbClientServiceTest.php]: sounds good 👍
